### PR TITLE
Decorate class

### PR DESCRIPTION
**Pull request by stephenkati**
- Implement `Decorator pattern design`
- Created the following classes:
1. `Nameable`
2. `BaseDecorator`
3. `CapitalizeDecorator`
4. `TrimmerDecorator`
- Established _inheritance_ in classes
- Implimented methods to:
- [ ] **capitalize** name
- [ ] **Trim** name to a max length of 10 characters

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -1,18 +1,25 @@
-class Person
+require_relative 'nameable'
```
Comment (Davidosky007) [inline person.rb:1]: * I get this error when i try to test your decorated class , Kindly fix this error 👇 and ensure you test to see if it works as expected 
image

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
Hi @stephenkati 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
* Well written PR 👍 
* Good looking readme 👍 
* No linter errors 👍 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @davidsoky007 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by ahmetbozaci** [comment]
### Hello @stephenkati 
### _This is an additional review as a result of the second opinion request._ 
- I am not getting any errors.
image



___
_**Please ask for another review in your Dashboard and the next Code Reviewer will mark it as approved right away**_



### Cheers and Happy coding! 💻 👏
____
[GitHub](https://github.com/ahmetbozaci) | [LinkedIn](https://linkedin.com/in/ahmetbozaci) | [Twitter](https://twitter.com/ahmtbozaci)

_Reviewed by [Ahmet Bozacı ](https://github.com/ahmetbozaci)_

**Review by Ibizugbe — APPROVED** [review]
Hi @stephenkati 👋👋, 

approve2


Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- [x] Linters are passing👍.
- [x] Everything works fine👍.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**